modules/runtime/nuxeo-common/src/main/java/org/nuxeo/common/xmap/registry/MapRegistry.java

```diff
@@ -81,6 +82,10 @@ public void initialize() {
     @Override
     protected void register(Context ctx, XAnnotatedObject xObject, Element element) {
         String id = (String) xObject.getRegistryId().getValue(ctx, element);
+        if (id == null) {
+            // prevent NPE on map key
+            id = "null";
```

Comment:
Ah good point, the concurrent maps disallow null keys...

Reply:
This was visible in tests (core bulk config declares a logConfig without id, maybe this should be changed)